CodeEdit/Features/Settings/Pages/SourceControlSettings/SourceControlGitView.swift

```diff
@@ -24,23 +29,61 @@ struct SourceControlGitView: View {
                 showMergeCommitsInPerFileLog
             }
         }
+        .onAppear {
+            Task {
+                authorName = try await gitConfig.get(key: "user.name", global: true) ?? ""
+                authorEmail = try await gitConfig.get(key: "user.email", global: true) ?? ""
+                preferRebaseWhenPulling = try await gitConfig.get(key: "pull.rebase", global: true) ?? false
+                Task {
+                    hasAppeared = true
+                }
+            }
+        }
     }
 }
 
 private extension SourceControlGitView {
     private var gitAuthorName: some View {
-        TextField("Author Name", text: $git.authorName)
+        TextField("Author Name", text: $authorName)
+            .onChange(of: authorName) { newValue in
+                if hasAppeared {
+                    Limiter.debounce(id: "authorNameDebouncer", duration: 0.5) {
+                        Task {
+                            await gitConfig.set(key: "user.name", value: newValue, global: true)
+                        }
+                    }
+                }
+            }
     }
 
     private var gitEmail: some View {
-        TextField("Author Email", text: $git.authorEmail)
+        TextField("Author Email", text: $authorEmail)
+            .onChange(of: authorEmail) { newValue in
+                if hasAppeared {
+                    Limiter.debounce(id: "authorEmailDebouncer", duration: 0.5) {
+                        Task {
+                            await gitConfig.set(key: "user.email", value: newValue, global: true)
+                        }
+                    }
+                }
+        }
     }
 
     private var preferToRebaseWhenPulling: some View {
         Toggle(
             "Prefer to rebase when pulling",
-            isOn: $git.preferRebaseWhenPulling
+            isOn: $preferRebaseWhenPulling
         )
+        .onChange(of: preferRebaseWhenPulling) { newValue in
+            if hasAppeared {
+                Limiter.debounce(id: "pullRebaseDebouncer", duration: 0.5) {
```

Comment:
I think the Limiter would have a nicer api if its async by default, so we dont have to specify Task every time this is called. We can also skip using Combine and use Task inside the Limiter class instead.

Reply:
Great idea!

Reply:
I just tried this, it wasn't debouncing properly doing it this way. This was my failed attempt..

```swift
enum Limiter {
    private static var throttleLastExecution: [AnyHashable: Date] = [:]
    private static var debounceTasks: [AnyHashable: Task<Void, Never>] = [:]

    /// Debounces an action, ensuring it is only called once after a specified delay if there are no other calls in the meantime.
    /// - Parameters:
    ///   - id: A unique identifier for the debounced action.
    ///   - duration: The debounce duration in seconds.
    ///   - action: The asynchronous action to be executed after the debounce period.
    static func debounce(id: AnyHashable, duration: TimeInterval, action: @escaping @Sendable () async -> Void) {
        // Cancel any existing task for the same ID
        debounceTasks[id]?.cancel()

        // Start a new debounced task
        debounceTasks[id] = Task {
            // Wait for the specified duration
            try? await Task.sleep(nanoseconds: UInt64(duration * 1_000_000_000))
            await action()

            // Remove the task from the dictionary once executed
            debounceTasks[id] = nil
        }
    }

    /// Throttles an action, ensuring it is only called once within the specified interval.
    /// - Parameters:
    ///   - id: A unique identifier for the throttled action.
    ///   - duration: The throttle duration in seconds.
    ///   - action: The asynchronous action to be executed after the throttle period.
    static func throttle(id: AnyHashable, duration: TimeInterval, action: @escaping @Sendable () async -> Void) {
        let now = Date()

        // Check if the action was recently executed
        if let lastExecution = throttleLastExecution[id], now.timeIntervalSince(lastExecution) < duration {
            return
        }

        // Update the last execution time and perform the action
        throttleLastExecution[id] = now
        Task {
            await action()
        }

        // Set up a timer to reset the throttle
        Task {
            try? await Task.sleep(nanoseconds: UInt64(duration * 1_000_000_000))
            throttleLastExecution[id] = nil
        }
    }
}

```

Reply:
Lets put a TODO on the code and I can revisit this in the future and update the API